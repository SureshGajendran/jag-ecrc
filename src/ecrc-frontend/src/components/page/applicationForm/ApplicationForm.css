.previousName {
  font-size: 18px;
  font-weight: 600;
  padding-left: 40px;
}

.note {
  font-size: 14px;
  font-weight: 400;
}
