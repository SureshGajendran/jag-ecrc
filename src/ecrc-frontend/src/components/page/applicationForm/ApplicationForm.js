import React, { useState } from "react";
import PropTypes from "prop-types";
import { useHistory } from "react-router-dom";

import "./ApplicationForm.css";
import Header from "../../base/header/Header";
import Footer from "../../base/footer/Footer";
import { SimpleForm } from "../../composite/simpleForm/SimpleForm";
import FullName from "../../composite/fullName/FullName";
import { Button } from "../../base/button/Button";
import SideCards from "../../composite/sideCards/SideCards";

export default function ApplicationForm({
  page: {
    header,
    applicant: {
      firstName,
      middleName,
      lastName,
      birthDate,
      sex,
      street,
      city,
      province,
      postalCode,
      country
    },
    setApplicant,
    org: { defaultScheduleTypeCd }
  }
}) {
  const [previousNames, setPreviousNames] = useState({
    previousTwo: false,
    previousThree: false
  });

  const [birthPlace, setBirthPlace] = useState("");
  const [birthPlaceError, setBirthPlaceError] = useState("");
  const [driversLicNo, setDriversLicNo] = useState("");
  const [phoneNumber, setPhoneNumber] = useState("");
  const [phoneNumberError, setPhoneNumberError] = useState("");
  const [emailAddress, setEmailAddress] = useState("");
  const [emailAddressError, setEmailAddressError] = useState("");
  const [jobTitle, setJobTitle] = useState("");
  const [jobTitleError, setJobTitleError] = useState("");
  const [organizationFacility, setOrganizationFacility] = useState("");
  const [organizationFacilityError, setOrganizationFacilityError] = useState(
    ""
  );

  const history = useHistory();

  const currentName = {
    firstName: {
      label: "First Name",
      id: "firstName",
      value: firstName,
      textInputStyle: "textinput_non_editable_gray"
    },
    middleName: {
      label: "Middle Name",
      id: "middleName",
      value: middleName,
      textInputStyle: "textinput_non_editable_gray"
    },
    lastName: {
      label: "Last Name",
      id: "lastName",
      value: lastName,
      textInputStyle: "textinput_non_editable_gray"
    }
  };

  const previousNameOne = {
    firstName: {
      label: "First Name",
      id: "previousFirstNameOne"
    },
    middleName: {
      label: "Middle Name",
      id: "previousMiddleNameOne"
    },
    lastName: {
      label: "Last Name",
      id: "previousLastNameOne"
    }
  };

  const previousNameTwo = {
    firstName: {
      label: "First Name",
      id: "previousFirstNameTwo"
    },
    middleName: {
      label: "Middle Name",
      id: "previousMiddleNameTwo"
    },
    lastName: {
      label: "Last Name",
      id: "previousLastNameTwo"
    }
  };

  const previousNameThree = {
    firstName: {
      label: "First Name",
      id: "previousFirstNameOneThree"
    },
    middleName: {
      label: "Middle Name",
      id: "previousMiddleNameThree"
    },
    lastName: {
      label: "Last Name",
      id: "previousLastNameThree"
    }
  };

  const applicantInformation = {
    title: null,
    textInputs: [
      {
        label: "City and Country of Birth",
        id: "birthPlace",
        isRequired: true,
        errorMsg: birthPlaceError,
        onChange: setBirthPlace
      },
      {
        label: "Date of Birth",
        id: "birthDate",
        value: birthDate,
        textInputStyle: "textinput_non_editable_gray"
      },
      {
        label: "Sex",
        id: "sex",
        value: sex,
        textInputStyle: "textinput_non_editable_gray"
      },
      {
        label: "BC Driver's Licence Number",
        id: "bcDLNumber",
        note: "(Current or Expired)",
        onChange: setDriversLicNo
      },
      {
        label: "Primary Phone Number",
        id: "phoneNumber",
        note: "(Include area code)",
        isRequired: true,
        errorMsg: phoneNumberError,
        onChange: setPhoneNumber
      },
      {
        label: "Personal Email Address",
        id: "emailAddress",
        note: "We may use this to communicate with you about your application.",
        isRequired: true,
        errorMsg: emailAddressError,
        onChange: setEmailAddress
      }
    ],
    buttons: []
  };

  const positionInformation = {
    title: "Position Information",
    textInputs: [
      {
        label: "Applicant's position/Job Title",
        id: "applicantPosition",
        isRequired: true,
        errorMsg: jobTitleError,
        onChange: setJobTitle
      }
    ],
    buttons: []
  };

  if (defaultScheduleTypeCd === "WBSD") {
    positionInformation.textInputs.push({
      label: "Organization Facility",
      id: "organizationFacility",
      note:
        "(Licenced Child Care Name, Adult Care Facility Name, or Contracted Company Name)",
      errorMsg: organizationFacilityError,
      onChange: setOrganizationFacility
    });
  }

  const address = {
    title: "Address",
    textInputs: [
      {
        label: "Street",
        id: "street",
        value: street,
        textInputStyle: "textinput_non_editable_gray"
      },
      {
        label: "City",
        id: "city",
        value: city,
        textInputStyle: "textinput_non_editable_gray"
      },
      {
        label: "Province",
        id: "province",
        value: province,
        textInputStyle: "textinput_non_editable_gray"
      },
      {
        label: "Postal Code",
        id: "postalCode",
        value: postalCode,
        textInputStyle: "textinput_non_editable_gray"
      },
      {
        label: "Country",
        id: "country",
        value: country,
        textInputStyle: "textinput_non_editable_gray"
      }
    ],
    buttons: []
  };

  const continueButton = {
    label: "Continue",
    buttonStyle: "btn btn-primary",
    buttonSize: "btn btn-sm",
    type: "submit"
  };

  const cancelButton = {
    label: "Cancel",
    buttonStyle: "btn btn-primary",
    buttonSize: "btn btn-sm",
    type: "submit"
  };

  const applicationVerification = () => {
    if (birthPlace === "") {
      setBirthPlaceError("Please enter a city and country of birth");
    }

    if (phoneNumber === "") {
      setPhoneNumberError("Please enter your primary phone number");
    }

    if (emailAddress === "") {
      setEmailAddressError("Please enter your personal email address");
    }

    if (jobTitle === "") {
      setJobTitleError("Please enter your position/job title");
    }

    if (defaultScheduleTypeCd === "WBSD" && organizationFacility === "") {
      setOrganizationFacilityError("Please enter your organization facility");
    }

    if (
      birthPlace !== "" &&
      phoneNumber !== "" &&
      emailAddress !== "" &&
      jobTitle !== "" &&
      !(defaultScheduleTypeCd === "WBSD" && organizationFacility === "")
    ) {
      setApplicant({
        firstName,
        middleName,
        lastName,
        birthDate,
        sex,
        street,
        city,
        province,
        postalCode,
        country,
        birthPlace,
        driversLicNo,
        phoneNumber,
        emailAddress,
        jobTitle,
        organizationFacility
      });

      history.push("/");
    }
  };

  const back = () => {
    history.push("/");
  };

  const additionalNames = event => {
    event.preventDefault();

    if (!previousNames.previousTwo) {
      setPreviousNames({ ...previousNames, previousTwo: true });
    } else if (!previousNames.previousThree) {
      setPreviousNames({ ...previousNames, previousThree: true });
    }
  };

  return (
    <main>
      <Header header={header} />
      <div className="page">
        <div className="content col-md-8">
          <h1>Criminal Record Check - Application</h1>
          <FullName title={"PERSONAL INFORMATION"} fullname={currentName} />
          <div className="heading">
            <span className="previousHeader">PREVIOUS NAME&nbsp;</span>
            <span className="note">
              Including alias, previous name and/or birthname
            </span>
          </div>
          <FullName title={null} fullname={previousNameOne} />
          {previousNames.previousTwo && (
            <FullName title={null} fullname={previousNameTwo} />
          )}
          {previousNames.previousThree && (
            <FullName title={null} fullname={previousNameThree} />
          )}
          {!previousNames.previousThree && (
            <span className="heading note previousFooter">
              If you have more than one previous name, please&nbsp;
              <button
                className="notAButton"
                type="button"
                onClick={event => additionalNames(event)}
              >
                click here to add them
              </button>
              .
            </span>
          )}
          <SimpleForm simpleForm={applicantInformation} />
          <SimpleForm simpleForm={positionInformation} />
          <SimpleForm simpleForm={address} />
          <div className="buttons">
            <Button button={cancelButton} onClick={back} />
            <Button button={continueButton} onClick={applicationVerification} />
          </div>
        </div>
        <div className="sidecard">
          <SideCards type={"personalinformation"} />
          <SideCards type={"collectionnotice"} />
        </div>
      </div>
      <Footer />
    </main>
  );
}

ApplicationForm.propTypes = {
  page: PropTypes.shape({
    header: PropTypes.shape({
      name: PropTypes.string.isRequired
    }),
    applicant: PropTypes.shape({
      firstName: PropTypes.string.isRequired,
      middleName: PropTypes.string.isRequired,
      lastName: PropTypes.string.isRequired,
      birthDate: PropTypes.string.isRequired,
      sex: PropTypes.string.isRequired,
      street: PropTypes.string.isRequired,
      city: PropTypes.string.isRequired,
      province: PropTypes.string.isRequired,
      postalCode: PropTypes.string.isRequired,
      country: PropTypes.string.isRequired
    }),
    setApplicant: PropTypes.func.isRequired,
    org: PropTypes.shape({
      defaultScheduleTypeCd: PropTypes.string.isRequired
    })
  }).isRequired
};
