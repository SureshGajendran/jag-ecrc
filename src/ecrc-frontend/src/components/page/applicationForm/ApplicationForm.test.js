import React from "react";
import { create } from "react-test-renderer";

import ApplicationForm from "./ApplicationForm";
import { MemoryRouter } from "react-router-dom";

describe("ApplicationForm Component", () => {
  test("Matches the snapshot", () => {
    const header = {
      name: "Criminal Record Check"
    };

    const applicant = {
      firstName: "Robert",
      middleName: "Norman",
      lastName: "Ross",
      birthPlace: "",
      birthDate: "1942-10-29",
      sex: "Male",
      bcDLNumber: "",
      phoneNumber: "",
      emailAddress: "",
      street: "123 Somewhere",
      city: "Here",
      province: "British Columbia",
      postalCode: "V9V 9V9",
      country: "Canada",
      applicantPosition: "",
      organizationFacility: ""
    };

    const org = {
      schedule: "D"
    };

    const page = {
      header,
      applicant,
      org
    };

    const applicationPage = create(
      <MemoryRouter>
        <ApplicationForm page={page} />
      </MemoryRouter>
    );
    expect(applicationPage.toJSON()).toMatchSnapshot();
  });
});
