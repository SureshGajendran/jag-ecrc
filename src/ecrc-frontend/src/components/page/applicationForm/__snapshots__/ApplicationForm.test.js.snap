// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationForm Component Matches the snapshot 1`] = `
<main>
  <header>
    <nav
      className="navbar navbar-expand-lg navbar-dark"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="https://www2.gov.bc.ca"
        >
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-none d-md-block"
            height="44"
            src="/images/bcid-logo-rev-en.svg"
            width="181"
          />
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-md-none"
            height="44"
            src="/images/bcid-symbol-rev.svg"
            width="64"
          />
        </a>
        <div
          className="navbar-brand"
        >
          Criminal Record Check
        </div>
        <button
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarNavAltMarkup"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarNavAltMarkup"
        >
          <div
            className="navbar-nav"
          >
            <a
              className="nav-item nav-link"
              href="/"
            >
              Home
            </a>
          </div>
        </div>
      </div>
    </nav>
  </header>
  <div
    className="page"
  >
    <div
      className="content col-md-8"
    >
      <h1>
        Criminal Record Check - Application
      </h1>
      <form
        className="simpleForm"
      >
        <div
          className="simpleForm_title"
        >
          PERSONAL INFORMATION
        </div>
        <ul>
          <li>
            <div>
              <label
                className="label"
                htmlFor="legalFirstNm"
              >
                <div
                  className="textinput_label"
                >
                  First Name
                </div>
                
                 
                <span
                  className="note"
                />
              </label>
              <input
                className="textinput_non_editable_gray"
                defaultValue="Robert"
                id="legalFirstNm"
                onChange={[Function]}
                readOnly={true}
                type="text"
              />
              <span
                className="error"
              />
            </div>
          </li>
          <li>
            <div>
              <label
                className="label"
                htmlFor="legalSecondNm"
              >
                <div
                  className="textinput_label"
                >
                  Middle Name
                </div>
                
                 
                <span
                  className="note"
                />
              </label>
              <input
                className="textinput_non_editable_gray"
                defaultValue="Norman"
                id="legalSecondNm"
                onChange={[Function]}
                readOnly={true}
                type="text"
              />
              <span
                className="error"
              />
            </div>
          </li>
          <li>
            <div>
              <label
                className="label"
                htmlFor="legalSurnameNm"
              >
                <div
                  className="textinput_label"
                >
                  Last Name
                </div>
                
                 
                <span
                  className="note"
                />
              </label>
              <input
                className="textinput_non_editable_gray"
                defaultValue="Ross"
                id="legalSurnameNm"
                onChange={[Function]}
                readOnly={true}
                type="text"
              />
              <span
                className="error"
              />
            </div>
          </li>
        </ul>
      </form>
      <div
        className="heading"
      >
        <span
          className="previousHeader"
        >
          PREVIOUS NAME 
        </span>
        <span
          className="note"
        >
          Including alias, previous name and/or birthname
        </span>
      </div>
      <form
        className="simpleForm"
      >
        <ul>
          <li>
            <div>
              <label
                className="label"
                htmlFor="alias1First"
              >
                <div
                  className="textinput_label"
                >
                  First Name
                </div>
                
                 
                <span
                  className="note"
                />
              </label>
              <input
                className="textinput_editable_white"
                defaultValue=""
                id="alias1First"
                onChange={[Function]}
                type="text"
              />
              <span
                className="error"
              />
            </div>
          </li>
          <li>
            <div>
              <label
                className="label"
                htmlFor="alias1Seocnd"
              >
                <div
                  className="textinput_label"
                >
                  Middle Name
                </div>
                
                 
                <span
                  className="note"
                />
              </label>
              <input
                className="textinput_editable_white"
                defaultValue=""
                id="alias1Seocnd"
                onChange={[Function]}
                type="text"
              />
              <span
                className="error"
              />
            </div>
          </li>
          <li>
            <div>
              <label
                className="label"
                htmlFor="alias1Surname"
              >
                <div
                  className="textinput_label"
                >
                  Last Name
                </div>
                
                 
                <span
                  className="note"
                />
              </label>
              <input
                className="textinput_editable_white"
                defaultValue=""
                id="alias1Surname"
                onChange={[Function]}
                type="text"
              />
              <span
                className="error"
              />
            </div>
          </li>
        </ul>
      </form>
      <span
        className="heading note previousFooter"
      >
        If you have more than one previous name, please 
        <button
          className="notAButton"
          onClick={[Function]}
          type="button"
        >
          click here to add them
        </button>
        .
      </span>
      <form
        className="simpleForm"
      >
        <ul>
          <li>
            <div>
              <label
                className="label"
                htmlFor="birthLoc"
              >
                <div
                  className="textinput_label"
                >
                  City and Country of Birth
                </div>
                <span
                  className="mandatory"
                  id="asterisk"
                >
                  *
                </span>
                 
                <span
                  className="note"
                />
              </label>
              <input
                className="textinput_editable_white"
                defaultValue=""
                id="birthLoc"
                onChange={[Function]}
                type="text"
              />
              <span
                className="error"
              >
                
              </span>
            </div>
          </li>
          <li>
            <div>
              <label
                className="label"
                htmlFor="birthDt"
              >
                <div
                  className="textinput_label"
                >
                  Date of Birth
                </div>
                
                 
                <span
                  className="note"
                />
              </label>
              <input
                className="textinput_non_editable_gray"
                defaultValue="1942-10-29"
                id="birthDt"
                onChange={[Function]}
                readOnly={true}
                type="text"
              />
              <span
                className="error"
              />
            </div>
          </li>
          <li>
            <div>
              <label
                className="label"
                htmlFor="genderTxt"
              >
                <div
                  className="textinput_label"
                >
                  Sex
                </div>
                
                 
                <span
                  className="note"
                />
              </label>
              <input
                className="textinput_non_editable_gray"
                defaultValue="Male"
                id="genderTxt"
                onChange={[Function]}
                readOnly={true}
                type="text"
              />
              <span
                className="error"
              />
            </div>
          </li>
          <li>
            <div>
              <label
                className="label"
                htmlFor="bcDLNumber"
              >
                <div
                  className="textinput_label"
                >
                  BC Driver's Licence Number
                </div>
                
                 
                <span
                  className="note"
                >
                  (Current or Expired)
                </span>
              </label>
              <input
                className="textinput_editable_white"
                defaultValue=""
                id="bcDLNumber"
                onChange={[Function]}
                type="text"
              />
              <span
                className="error"
              />
            </div>
          </li>
          <li>
            <div>
              <label
                className="label"
                htmlFor="phoneNumber"
              >
                <div
                  className="textinput_label"
                >
                  Primary Phone Number
                </div>
                <span
                  className="mandatory"
                  id="asterisk"
                >
                  *
                </span>
                 
                <span
                  className="note"
                >
                  (Include area code)
                </span>
              </label>
              <input
                className="textinput_editable_white"
                defaultValue=""
                id="phoneNumber"
                onChange={[Function]}
                placeholder="123 456 7890"
                type="text"
              />
              <span
                className="error"
              >
                
              </span>
            </div>
          </li>
          <li>
            <div>
              <label
                className="label"
                htmlFor="emailAddress"
              >
                <div
                  className="textinput_label"
                >
                  Personal Email Address
                </div>
                <span
                  className="mandatory"
                  id="asterisk"
                >
                  *
                </span>
                 
                <span
                  className="note"
                >
                  We may use this to communicate with you about your application.
                </span>
              </label>
              <input
                className="textinput_editable_white"
                defaultValue=""
                id="emailAddress"
                onChange={[Function]}
                placeholder="example@test.com"
                type="text"
              />
              <span
                className="error"
              >
                
              </span>
            </div>
          </li>
        </ul>
      </form>
      <form
        className="simpleForm"
      >
        <div
          className="simpleForm_title"
        >
          Position Information
        </div>
        <ul>
          <li>
            <div>
              <label
                className="label"
                htmlFor="applicantPosition"
              >
                <div
                  className="textinput_label"
                >
                  Applicant's position/Job Title
                </div>
                <span
                  className="mandatory"
                  id="asterisk"
                >
                  *
                </span>
                 
                <span
                  className="note"
                />
              </label>
              <input
                className="textinput_editable_white"
                defaultValue=""
                id="applicantPosition"
                onChange={[Function]}
                type="text"
              />
              <span
                className="error"
              >
                
              </span>
            </div>
          </li>
          <li>
            <div>
              <label
                className="label"
                htmlFor="organizationFacility"
              >
                <div
                  className="textinput_label"
                >
                  Organization Facility
                </div>
                <span
                  className="mandatory"
                  id="asterisk"
                >
                  *
                </span>
                 
                <span
                  className="note"
                >
                  (Licenced Child Care Name, Adult Care Facility Name, or Contracted Company Name)
                </span>
              </label>
              <input
                className="textinput_editable_white"
                defaultValue=""
                id="organizationFacility"
                onChange={[Function]}
                type="text"
              />
              <span
                className="error"
              >
                
              </span>
            </div>
          </li>
        </ul>
      </form>
      <form
        className="simpleForm"
      >
        <div
          className="simpleForm_title"
        >
          Current Street Address
        </div>
        <ul>
          <li>
            <div>
              <label
                className="label"
                htmlFor="addressLine1"
              >
                <div
                  className="textinput_label"
                >
                  Street
                </div>
                
                 
                <span
                  className="note"
                />
              </label>
              <input
                className="textinput_non_editable_gray"
                defaultValue="123 Somewhere"
                id="addressLine1"
                onChange={[Function]}
                readOnly={true}
                type="text"
              />
              <span
                className="error"
              />
            </div>
          </li>
          <li>
            <div>
              <label
                className="label"
                htmlFor="cityNm"
              >
                <div
                  className="textinput_label"
                >
                  City
                </div>
                
                 
                <span
                  className="note"
                />
              </label>
              <input
                className="textinput_non_editable_gray"
                defaultValue="Here"
                id="cityNm"
                onChange={[Function]}
                readOnly={true}
                type="text"
              />
              <span
                className="error"
              />
            </div>
          </li>
          <li>
            <div>
              <label
                className="label"
                htmlFor="provinceNm"
              >
                <div
                  className="textinput_label"
                >
                  Province
                </div>
                
                 
                <span
                  className="note"
                />
              </label>
              <input
                className="textinput_non_editable_gray"
                defaultValue="British Columbia"
                id="provinceNm"
                onChange={[Function]}
                readOnly={true}
                type="text"
              />
              <span
                className="error"
              />
            </div>
          </li>
          <li>
            <div>
              <label
                className="label"
                htmlFor="postalCodeTxt"
              >
                <div
                  className="textinput_label"
                >
                  Postal Code
                </div>
                
                 
                <span
                  className="note"
                />
              </label>
              <input
                className="textinput_non_editable_gray"
                defaultValue="V9V 9V9"
                id="postalCodeTxt"
                onChange={[Function]}
                readOnly={true}
                type="text"
              />
              <span
                className="error"
              />
            </div>
          </li>
          <li>
            <div>
              <label
                className="label"
                htmlFor="countryNm"
              >
                <div
                  className="textinput_label"
                >
                  Country
                </div>
                
                 
                <span
                  className="note"
                />
              </label>
              <input
                className="textinput_non_editable_gray"
                defaultValue="Canada"
                id="countryNm"
                onChange={[Function]}
                readOnly={true}
                type="text"
              />
              <span
                className="error"
              />
            </div>
          </li>
        </ul>
      </form>
      <p>
        Is your mailing address different from your current street address? 
        <input
          onClick={[Function]}
          type="checkbox"
        />
      </p>
      <div
        className="buttons"
      >
        <button
          className="btn btn-primary btn btn-sm"
          onClick={[Function]}
          type="submit"
        >
          Cancel
        </button>
        <button
          className="btn btn-primary btn btn-sm"
          onClick={[Function]}
          type="submit"
        >
          Continue
        </button>
      </div>
    </div>
    <div
      className="sidecard"
    >
      <div>
        <div
          className="dashboard-spacing"
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="row"
          >
            <div
              className="col-12"
              style={
                Object {
                  "paddingTop": "30px",
                  "position": "relative",
                }
              }
            >
              <p />
              <section
                className="submit-container"
                id="blue-section"
              >
                <h2
                  className="heading-style"
                >
                  Update your personal information
                </h2>
                <div
                  className="submit-content"
                >
                  <span>
                    If you find an error in your personal information, please contact 
                    <a
                      className="link"
                      href="https://www2.gov.bc.ca/gov/content/governments/organizational-structure/ministries-organizations/ministries/citizens-services/servicebc"
                    >
                      Service BC
                    </a>
                     ,
                    <a
                      className="link"
                      href="https://www.icbc.com/Pages/default.aspx"
                    >
                      ICBC
                    </a>
                     or 
                    <a
                      className="link"
                      href="https://www.addresschange.gov.bc.ca/"
                    >
                      AddressChangeBC
                    </a>
                     to correct it.
                  </span>
                </div>
              </section>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          className="wide-dashboard-spacing"
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="row"
          >
            <div
              className="col-12"
              style={
                Object {
                  "paddingTop": "30px",
                  "position": "relative",
                }
              }
            >
              <p />
              <section
                className="submit-container"
                id="notice-section"
                style={
                  Object {
                    "backgroundColor": "#F2F2F2",
                    "border": "none",
                    "color": "#000",
                  }
                }
              >
                <h2
                  style={
                    Object {
                      "color": "#000",
                    }
                  }
                >
                  Collection Notice
                </h2>
                <div
                  className="submit-content"
                >
                  <span>
                    <div
                      style={
                        Object {
                          "fontSize": "12px",
                        }
                      }
                    >
                      The Security Programs Division will collect your personal information for the purpose of fulfilling the criminal record check requirements of the Criminal Records Review Act (CRRA) and in accordance with sections 26(a) and (c) of the Freedom of Information and Protection of Privacy Act (FoIPPA) and section 6 of the CRRA . Additionally, SPD may collect personal information under section 26(e) of FoIPPA for the purpose of evaluating the Criminal Records Review Program and activities to better serve you. Should you have any questions about the collection, use, or disclosure of personal information, please contact the Criminal Records Review Program, Security Programs Division via mail to PO Box 9217 Stn Prov Govt Victoria, BC V8W 9J1; email to 
                      <a
                        href="mailto:criminalrecords@gov.bc.ca"
                      >
                        criminalrecords@gov.bc.ca
                      </a>
                       or by telephone at 1- 855-587-0185 (option 2).
                    </div>
                  </span>
                </div>
              </section>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="footer"
    style={Object {}}
  >
    <nav
      className="navbar navbar-expand-lg navbar-dark"
    >
      <div
        className="container"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/disclaimer"
              rel="noopener noreferrer"
              target="_blank"
            >
              Disclaimer
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/privacy"
              rel="noopener noreferrer"
              target="_blank"
            >
              Privacy
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/accessibility"
              rel="noopener noreferrer"
              target="_blank"
            >
              Accessibility
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/copyright"
              rel="noopener noreferrer"
              target="_blank"
            >
              Copyright
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/contact-us"
              rel="noopener noreferrer"
              target="_blank"
            >
              Contact Us
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </footer>
</main>
`;
